### ISSUE TITLE
Deserialize empty protobuf list on iOS 

### ISSUE THREAD

**Issue by webfrea-k** [Kotlin/kotlinx.serialization#355]
I have the following data class:
```
@Serializable
data class Test (
        @Optional @SerialId(1) var events: List<Event> = emptyList(),
        @Optional @SerialId(2) var games: List<Game> = emptyList()
)
```
Protobuf deserialization (proto3) works correct on Android and iOS if both values are populated. If at least one list property is empty (empty list), the deserialization fails on iOS with error: `Unexpected EOF`.

If empty property is commented out of data class, deserialization works on iOS as well.

**Comment by sandwwraith** [comment]
Hi, could you please give a sample of protobuf bytes so I could debug this? And definitions of `Event` and `Game` classes, too

**Comment by webfrea-k** [comment]
Unfortunately the only example producing this error has 28k+ bytes in the response (and the entire `Test` class is way bigger than just `events` and `games` too). But maybe my observations can help:

I've implemented custom deserializer for this class and debug the following while loop:

```
loop@ while (true) {
        when (val i = inp.decodeElementIndex(descriptor)) { ... }
}
inp.endStructure(descriptor)
```
Is it possible, that `decodeElementIndex` is somehow not incremented if the `repeated` field is empty and while loop eventually runs until `EOF` is thrown?

**Comment by sandwwraith** [comment]
Hm, do you use `ktor` client on ios? It could be possibly related to #360 , since errors occur only on big data

**Comment by webfrea-k** [comment]
You are correct! I managed to make the response shorter and data is indeed handled correctly on iOS as well.
- Max length working on both platforms (that I can re-produce): 96735 bytes
- Length working on android and not on iOS:  128153 bytes
